src/app/routines/routineCard/routineOptionsButton/RoutineOptionsButton.tsx

```diff
@@ -1,14 +1,42 @@
+import { deleteRoutine } from '@/app/api/routineApi';
+import ConfirmationDialog from '@/app/components/dialogs/ConfirmationDialog';
+import { AppRoutes } from '@/types/routes';
 import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
 import EditIcon from '@mui/icons-material/Edit';
 import MoreVertIcon from '@mui/icons-material/MoreVert';
 import { Divider, IconButton, ListItemIcon, Menu, MenuItem, Tooltip, Typography } from '@mui/material';
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import './RoutineOptionsButton.scss';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
 
-const RoutineOptionsButton = () => {
+const RoutineOptionsButton: React.FC<Props> = ({ routineId }) => {
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
   const open = Boolean(anchorEl);
+  const navigate = useNavigate();
+  const [openConfirmationDialog, setOpenConfirmationDialog] = useState(false);
+  const queryClient = useQueryClient();
+
+  const { mutateAsync: deleteRoutineMutation } = useMutation({
+    mutationFn: deleteRoutine,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ['routines'] });
+    },
+  });
+
+  const handleClose = () => {
+    setOpenConfirmationDialog(false);
+  };
+
+  const handleConfirm = async () => {
+    try {
```

Comment:
I think that this error check may be redundant sincereact-query already provides error handling.
I suggest the following approach instead (what do you think):
```
const deleteRoutineMutation = useMutation(deleteRoutine, {
  onSuccess: () => {
    // Close the dialog on success
    setOpenConfirmationDialog(false);
  },
  onError: (error) => {
    // Handle the error, such as logging or showing an alert
    console.error("Error deleting routine:", error);
  },
});

...
const handleConfirm = () => {
  deleteRoutineMutation.mutate(routineId);
};
```